### PR TITLE
Delete student module command

### LINKED ISSUE #22: As a student, I can delete which modules my friend is taking this semester
## Delete which Modules my Friend is taking
Remove a module from a friend in the application
delete_module [friend_index] [module_name] 
friend_index: valid index taken from the GUI list
module_name: “<faculty_code><module_number>”

### PR DESCRIPTION
**Pull request by blaukc**
Implemented delete_module command (delete_module i/INDEX m/MODULECODE)

Added some SLAP to add_module as well 

Closes: https://github.com/AY2324S2-CS2103T-W12-2/tp/issues/22

### CODE REVIEW

**Review by taufiq — APPROVED** [review]
solid

File: src/main/java/seedu/address/model/AddressBook.java [hunk 1]
```diff
@@ -108,12 +108,32 @@ public void addModuleToStudent(ModuleCode m, Student s) {
...
                 return;
             }
         }
     }
+
+    /**
+     * Adds a module to a student in the address book.
+     *
+     * @param m The module code to be added.
+     * @param s The student to whom the module is to be added.
```
Comment (yadunut) [inline src/main/java/seedu/address/model/AddressBook.java:122]: The comments on this function are wrong

File: src/main/java/seedu/address/model/student/Student.java [hunk 2]
```diff
@@ -82,6 +82,25 @@ public List<ModuleCode> getModules() {
...
+     * @return true if module is taken by student
+     */
+    public boolean addModule(ModuleCode m) {
+        return modules.add(m);
+    }
+
+    /**
+     * Checks if module `m` is held within the student
+     * @param m Module to check
+     * @return true if module is taken by student
```
Comment (yadunut) [inline src/main/java/seedu/address/model/student/Student.java:98]: Comments are wrong

File: src/main/java/seedu/address/model/student/Student.java [hunk 3]
```diff
@@ -82,6 +82,25 @@ public List<ModuleCode> getModules() {
     public boolean hasModule(ModuleCode m) {
         return modules.contains(m);
     }
+
+    /**
+     * Checks if module `m` is held within the student
+     * @param m Module to check
+     * @return true if module is taken by student
```
Comment (yadunut) [inline src/main/java/seedu/address/model/student/Student.java:89]: Comments are wrong

**Pull request merged**